cmd/sidecar-proxy/main.go

```diff
@@ -122,18 +123,34 @@ func (pi *ProxyInstance) Start(proxyHost, access, refresh string) error {
 		}
 	}
 
-	pi.log.Printf("Listening on %s", listenAddr)
-	pi.svr = &http.Server{
-		Addr:      listenAddr,
-		Handler:   pi.Handler(proxyURL, access, refresh),
-		TLSConfig: pi.TLSConfig,
+	portNum, err := strconv.Atoi(port)
+	if err != nil {
+		return err
 	}
 
-	if err := pi.svr.ListenAndServeTLS("", ""); err != nil {
-		fmt.Printf("error: %+v\n", err)
-		return err
+	var retryListenAndServeTLS func(int) error
+	retryListenAndServeTLS = func(port int) error {
+		listenAddr := fmt.Sprintf(":%v", strconv.Itoa(port))
+		pi.log.Printf("Listening on %s", listenAddr)
+		pi.svr = &http.Server{
+			Addr:      listenAddr,
+			Handler:   pi.Handler(proxyURL, access, refresh),
+			TLSConfig: pi.TLSConfig,
+		}
+
+		if err := pi.svr.ListenAndServeTLS("", ""); err != nil {
+			var optErr *net.OpError
+			if errors.As(err, &optErr) {
+				if optErr.Op == "listen" && strings.Contains(optErr.Error(), "address already in use") {
+					return retryListenAndServeTLS(port + 1)
```

Comment:
Just wondering about the log output here when there is a conflict, e.g. if port 9000 is already in use, would we see the following deceptive log messages:

```
Listening on :9000
Listening on :9001
```

Reply:
Updated.